### DISCUSSION TITLE
Using Zksync on remix

### DISCUSSION THREAD

**Discussion by Donpele01** [Cyfrin/foundry-full-course-cu#1902] [Q&A] [answered]
Zksync is not compiling on my remix as shown in the tutorial but shows a green check mark for compilation and then this message on the deploy tab: No contracts ready for deployment yet, compile a solidity contract.

**Comment by rangoc** [comment]
> Zksync is not compiling on my remix as shown in the tutorial but shows a green check mark for compilation and then this message on the deploy tab: No contracts ready for deployment yet, compile a solidity contract.

I also experienced this issue. Another user who experienced this same issue, has opened a issue on zksync's plugin repo
https://github.com/NethermindEth/zksync-remix-plugin/issues/124

So maybe be on a lookout for a solution in that discussion ☝🏻

**Reply by EngrPips** [reply]
> > Zksync is not compiling on my remix as shown in the tutorial but shows a green check mark for compilation and then this message on the deploy tab: No contracts ready for deployment yet, compile a solidity contract.
> 
> I also experienced this issue. Another user who experienced this same issue, has opened a issue on zksync's plugin repo [NethermindEth/zksync-remix-plugin#124](https://github.com/NethermindEth/zksync-remix-plugin/issues/124)
> 
> So maybe be on a lookout for a solution in that discussion ☝🏻


Thanks for the help, Friend.

**Reply by Donpele01** [reply]
> > Zksync is not compiling on my remix as shown in the tutorial but shows a green check mark for compilation and then this message on the deploy tab: No contracts ready for deployment yet, compile a solidity contract.
> 
> I also experienced this issue. Another user who experienced this same issue, has opened a issue on zksync's plugin repo [NethermindEth/zksync-remix-plugin#124](https://github.com/NethermindEth/zksync-remix-plugin/issues/124)
> 
> So maybe be on a lookout for a solution in that discussion ☝🏻

Thanks, i just checked the post out.

**Reply by GuireWire** [reply]
i was going crazy. i thought it was just me! Because i went back over the courses 1-3 after the update and I noticed how it wouldn't deploy

**Reply by EngrPips** [reply]
It is not only you, Friend.

**Reply by BowtiedHarpyEagle** [reply]
> > > Zksync is not compiling on my remix as shown in the tutorial but shows a green check mark for compilation and then this message on the deploy tab: No contracts ready for deployment yet, compile a solidity contract.
> > 
> > 
> > I also experienced this issue. Another user who experienced this same issue, has opened a issue on zksync's plugin repo [NethermindEth/zksync-remix-plugin#124](https://github.com/NethermindEth/zksync-remix-plugin/issues/124)
> > So maybe be on a lookout for a solution in that discussion ☝🏻
> 
> Thanks, i just checked the post out.

I have the same issue. Following both this thread and the nethermind zksync repo. Hope it gets resolved soon.

**Reply by EngrPips** [reply]
I am looking forward to the solution myself.

**Comment by Verifieddanny** [comment, marked as answer]
I used ATLAS IDE to deploy mine since it's not working on Remix

https://zksync.atlaszk.com/ide

**Reply by EngrPips** [reply]
Thanks for sharing this. Friend. I should check this out.

**Comment by marcinczenko** [comment]
There is an update published in [Lesson 25: Zksync Plugin Fix](https://updraft.cyfrin.io/courses/solidity/fund-me/zksync-plugin-fix), which says that the solution is to put the contract file inside the top-level `contracts` folder. This worked for me well and now I  have the _Deploy_ section of the Remix Zksync plugin showing the compiled contract being ready to deploy.

**Reply by EngrPips** [reply]
Awesome, I am glad that fix works for you. Keep crushing it, buddy.

**Reply by BowtiedHarpyEagle** [reply]
Issue is now fixed with plugin version 0.4.2

**Reply by EngrPips** [reply]
Interesting.
